### DISCUSSION TITLE
R - penggunaan across - Mengapa ada gagal menjalankan fungsi ketika tidak menggunakan tilde?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#104] [Q&A] [answered]
Jadi saya ingin mengubah NA menjadi 0 dengan menggunakan mutate + across, 
menurut sepemahaman terbatas saya, penerapannya dengan memasukan fungsi setelah pemilihan data tujuan di dalam kurung bersama across, 
namun ternyata gagal berjalan
![without tilde does not work](https://user-images.githubusercontent.com/95233522/168343606-1cc5b4cc-7a31-45c5-bb76-fefe309d08de.JPG)

Namun setelah mencari-cari solusi, ternyata bisa tinggal ditambah tilde (~), tapi saya belum paham apa bedanya? 
![with tilde it works](https://user-images.githubusercontent.com/95233522/168343615-c247be15-499c-4e9e-97bf-a1c84bf24dfc.JPG)

Kapan kita perlu pakai tilde? Kapan tidak perlu pakai tilde? Karena sebelumnya, saya pakai mutate + across untuk mengubah beberapa kolom menjadi numeric bisa tanpa tilde.
![numeric tanpa tilde](https://user-images.githubusercontent.com/95233522/168344170-3481055e-6f80-490e-b695-b70beed0407f.JPG)

**Comment by inytss** [comment, marked as answer]
Dear @Hertzspr ,
Mengenai penggunaan tilde(~) pada fungsi `across()`, saya coba bantu untuk menjelaskan secara sederhana:
- tanpa tilde: biasanya digunakan ketika penggunaan `across()` dan fungsi yang akan dilakukan perubahan pada kolom-kolom tersebut adalah fungsi yang diterapkan langsung pada kolom tersebut, seperti pengubahan tipe data. Kolom tersebut yang langsung terganti tanpa harus memanggil parameter dalam fungsi `as.numeric()`
- menggunakan tilde: biasanya digunakan ketika fungsi yang akan diterapkan pada `across()` adalah fungsi yang mengakses value dari suatu kolom, seperti `replace_na()` yang mana nilai 0 pada `replace_na()` mengisi suatu parameter dalam fungsi tersebut. Sehingga dapat disimpulkan ketika ingin menggunakan fungsi `across()` dan parameter `.fns = ` adalah fungsi untuk melakukan perubahan pada nilai dalam kolom yang mengakses parameter fungsi tersebut, maka perlu menggunakna tilde.

Semoga menjawab :)
